lib/processors/jsdoc/lib/ui5/plugin.js

```diff
@@ -516,6 +523,190 @@ function isPotentialEnum(node) {
+				const potentialRenaming = checkVarRenaming(curVar);
+				const filteredValues = curVar.references
+					.filter((ref) => ref.writeExpr)
+					.map((ref) => {
+						const curNode = ref.writeExpr;						
+						if ( curNode.type === Syntax.MemberExpression && !curNode.computed && curNode.object.type === Syntax.Identifier ) {
+							return curNode.object.name;
+						} else if (curNode.type === Syntax.MemberExpression && curNode.object.type === Syntax.MemberExpression) { // Standalone variable without leading dot notation namespace
+							return getResolvedObjectName(curNode); 
+						}
+
+						return (curNode && curNode.name) || "";
+					});
+
+				acc.push({
+					ref: filteredValues[0],
+					original: potentialRenaming
+						? potentialRenaming.original
+						: curVar.name,
+					renamed: potentialRenaming ? potentialRenaming.renamed : null,
+					replaced: potentialRenaming ? potentialRenaming.replaced : null,
+				});
+				return acc;
+			}, [])[0];
+
+		leftMostName = potentialChunk && potentialChunk.ref;
+
+		if (potentialChunk && potentialChunk.replaced) {
+			originalName = originalName.replace(potentialChunk.replaced, potentialChunk.original);
+		} else {
+			originalName = leftMostName
+				? leftMostName + "." + originalName
+				: originalName;
+		}
+
+		if (potentialChunk && potentialChunk.renamed) {
+			originalName = originalName.replace(
+				potentialChunk.renamed + ".",
+				potentialChunk.original + "."
+			);
+		}
+	}
+
+	// When an enum is defined in an external module, we cannot rely that this module is loaded and we could
+	// read from enumValues variable. Therefore we need to resolve the enum name from the type. 
+	// The imported dependency name would be replaced by the type i.e. IF type: "sap.ui.core" THEN coreLibrary -> sap.ui.core
+	// The leftMostChunk is the (eventual) dependency name i.e. coreLibrary
+	let originalNameChunks = originalName.split(".");
+	let typeChunks = type.split(".")
+	const firstMatchedChunkIndex = originalNameChunks.findIndex((chunk) => typeChunks.indexOf(chunk) !== -1);
+
+ 	if (firstMatchedChunkIndex !== -1) {
+		originalNameChunks = originalNameChunks.slice(firstMatchedChunkIndex);
+		const typeIndex = typeChunks.indexOf(originalNameChunks[0]);
+		typeChunks = typeChunks.slice(0, typeIndex);
+		originalNameChunks = typeChunks.concat(originalNameChunks);
+	}
+
+	return originalNameChunks;
+}
+
+/**
+ * Gets enclosing scope
+ * 
+ * @param {Node} node 
+ * @returns {Scope}
+ */
+ function getEnclosingVariableScope (node) {
+	// Get to the nearest upper scope
+	let nearestScopeableNode = node;
+	while (
+		nearestScopeableNode && nearestScopeableNode.type &&
+		nearestScopeableNode.type !== Syntax.ArrowFunctionExpression &&
+		nearestScopeableNode.type !== Syntax.FunctionDeclaration &&
+		nearestScopeableNode.type !== Syntax.FunctionExpression
+	) {
+		nearestScopeableNode = nearestScopeableNode.parent;
+	}
+
+	const scopeManager = escope.analyze(currentProgram);
```

Comment:
This analyse step ideally should be done only once. Either lazy on first access or when the currentProgram is set.

Reply:
Now it should be ok. Setting it up where the currentProgram is being set